tuist/tuist
# Automatically generate target dependencies

### Short description 📝

This PR adds a new `TargetDependency`, `.auto`. If this is present, during generation dependencies will be added to the target based on `import` statements in the source files. For now, only `.target` dependencies are added.

This is a work-in-progress. At this point, I'm especially looking for feedback into how the user will enable this option. I'm leaning towards changing this from a `TargetDependency` to a generation option. Ideally, it will be possible to specify this on both a per-project and per-target basis.

### How to test the changes locally 🧐

`Project.swift` has been changed to use this option, so existing tests should verify that things haven't broken. Additionally, I have modified the `ios_app_with_tests` fixture to use it.

### Checklist ✅
